autoload/translate.vim

```diff
@@ -43,6 +43,38 @@ function! translate#replace(source_target) range abort
   let @a = l:backup
 endfunction
 
+function! translate#wrap_operator_show(type)
+    call translate#operator(a:type, 0)
+endfunction
+
+function! translate#wrap_operator_replace(type)
+    call translate#operator(a:type, 1)
+endfunction
+
+function! translate#operator(type, replace) abort
```

Comment:
What about making the `replace` param optional like so
```vimscript
function! translate#operator(type, ...) abort
  if !s:check_executable() | return | endif

  let l:replace = get(a:, 1, 0)
```

So we can directly use it there https://github.com/VincentCordobes/vim-translate/blob/0dbd121c67b26ce781ac906355ac6516885155b4/plugin/translate.vim#L12

And get ride of https://github.com/VincentCordobes/vim-translate/blob/0dbd121c67b26ce781ac906355ac6516885155b4/autoload/translate.vim#L46-L48

Reply:
Hmm, I think the two wrapper functions are needed to distinguish between the two `operatorfunc` (replace or not replace), because we can't give it an argument. We wouldn't be able to tell the `<Plug>TranslateReplace` mapping that it needs to call `operator()` with another argument.

I tried to use lambdas or function variables but they aren't real functions apparently

Reply:
Yes for sure `translate#replace_operator` is needed. Yet `translate#wrap_operator_show` is not @perelo

Reply:
Oh ok, there is one `replace_operator()` and one `operator()` :+1: :+1: